src/index.ts

```diff
@@ -152,293 +168,288 @@ export default function rehypePrettyCode(
+          codeTree = hastParser.parse(
+            highlighter.codeToHtml(strippedValue, getOptions('plaintext')),
+          );
         }
 
-        Object.entries(trees).forEach(([, tree]) => {
-          let lineCounter = 0;
+        let lineCounter = 0;
 
-          const wordOptions: CharsHighlighterOptions = {
-            ranges: wordNumbers,
-            idsMap: wordIdsMap,
-            counterMap: new Map(),
-          };
+        const charsHighlighterOptions: CharsHighlighterOptions = {
+          ranges: charsListNumbers,
+          idsMap: charsListIdMap,
+          counterMap: new Map<string, number>(),
+        };
 
-          visit(tree, 'element', (element) => {
-            if (
-              element.tagName === 'code' &&
-              /srebmuNeniLwohs(?!(.*)(\/))/.test(reverseString(meta))
-            ) {
+        visit(codeTree, 'element', (element) => {
+          if (
+            element.tagName === 'code' &&
+            /srebmuNeniLwohs(?!(.*)(\/))/.test(reverseString(meta))
+          ) {
+            if (element.properties) {
+              element.properties['data-line-numbers'] = '';
+            }
+
+            const lineNumbersStartAtMatch = reverseString(meta).match(
+              /(?:\}(\d+){)?srebmuNeniLwohs(?!(.*)(\/))/,
+            );
+            const startNumberString = lineNumbersStartAtMatch?.[1];
+            if (startNumberString) {
+              const startAt = startNumberString
+                ? Number(reverseString(startNumberString)) - 1
+                : 0;
+              lineNumbersMaxDigits = startAt;
               if (element.properties) {
-                element.properties['data-line-numbers'] = '';
+                element.properties.style = `counter-set: line ${startAt};`;
               }
+            }
+          }
 
-              const lineNumbersStartAtMatch = reverseString(meta).match(
-                /(?:\}(\d+){)?srebmuNeniLwohs(?!(.*)(\/))/,
-              );
-              const startNumberString = lineNumbersStartAtMatch?.[1];
-              if (startNumberString) {
-                const startAt = startNumberString
-                  ? Number(reverseString(startNumberString)) - 1
-                  : 0;
-                lineNumbersMaxDigits = startAt;
-                if (element.properties) {
-                  element.properties.style = `counter-set: line ${startAt};`;
-                }
-              }
+          if (
+            Array.isArray(element.properties?.className) &&
+            element.properties?.className?.[0] === 'line'
+          ) {
+            if (grid && element.children.length === 0) {
+              element.children = [{ type: 'text', value: ' ' }];
             }
 
+            replaceLineClass(element);
+            onVisitLine?.(element);
+
             if (
-              Array.isArray(element.properties?.className) &&
-              element.properties?.className?.[0] === 'line'
+              lineNumbers.length !== 0 &&
```

Comment:
```suggestion
```
Can be removed since next check `includes` will return `false` for empty array `lineNumbers`